File: src/main/java/me/pcasaes/hexoids/core/domain/index/BarrierSpatialIndexFactory.java
package me.pcasaes.hexoids.core.domain.index;

public class BarrierSpatialIndexFactory {

    private static final BarrierSpatialIndexFactory FACTORY = new BarrierSpatialIndexFactory();

    private BarrierSpatialIndexFactory() {
    }

    public static BarrierSpatialIndexFactory factory() {
        return FACTORY;
    }

    private BarrierSpatialIndex barrierSpatialIndex;

    public BarrierSpatialIndex get() {
        return barrierSpatialIndex;
    }

    public void setBarrierSpatialIndex(BarrierSpatialIndex barrierSpatialIndex) {
        this.barrierSpatialIndex = barrierSpatialIndex;
    }
}

File: src/main/kotlin/me/pcasaes/hexoids/core/domain/index/BarrierSpatialIndexFactory.kt
package me.pcasaes.hexoids.core.domain.index

object BarrierSpatialIndexFactory {

    private lateinit var barrierSpatialIndex: BarrierSpatialIndex

    fun get(): BarrierSpatialIndex {
        return barrierSpatialIndex
    }

    fun setBarrierSpatialIndex(barrierSpatialIndex: BarrierSpatialIndex) {
        this.barrierSpatialIndex = barrierSpatialIndex
    }
}
